### PR TITLE
provision service from CRD

### PR DESCRIPTION
**Pull request by philbrookes**

### CODE REVIEW

File: deploy/SharedService_crd.yaml [hunk 1]
```diff
@@ -18,3 +18,12 @@ spec:
           properties:
             image:
               type: string
+            clusterServiceClassExternalName:
+              type: string
+            clusterServiceClassName:
```
Comment (maleck13) [inline deploy/SharedService_crd.yaml:23]: do we need both of these can one be found out from the other

File: pkg/apis/aerogear/v1alpha1/types.go [hunk 2]
```diff
@@ -23,15 +23,19 @@ type SharedService struct {
 
 type SharedServiceSpec struct {
```
Comment (maleck13) [inline pkg/apis/aerogear/v1alpha1/types.go:24]: So I actually think this should be called a shared service config and each time you create an instance based on the config it creates a shared service with maxProvisions etc. But I can iterate on this

**Review by maleck13 — APPROVED** [review]
gonna merge

File: deploy/SharedService_crd.yaml (hunk 1 above)
Comment (philbrookes) [reply deploy/SharedService_crd.yaml:23]: This is to allow the creator of this CRD to add a human-friendly external name, but that is slower to look up, so after the service class name is found it is stored in the CRD for future faster lookups.

**Pull request merged**